targets/simple_switch/register_access.h

```diff
@@ -0,0 +1,110 @@
+/* Copyright 2019-present Cisco Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Andy Fingerhut (jafinger@cisco.com)
+ *
+ */
+
+#ifndef SIMPLE_SWITCH_REGISTER_ACCESS_H_
+#define SIMPLE_SWITCH_REGISTER_ACCESS_H_
+
+
+class RegisterAccess {
+ public:
+    static constexpr int PACKET_LENGTH_REG_IDX = 0;
+
+    static constexpr int CLONE_MIRROR_SESSION_ID_REG_IDX = 1;
+    static constexpr uint64_t CLONE_MIRROR_SESSION_ID_MASK = 0x000000000000ffff;
+    static constexpr uint64_t CLONE_MIRROR_SESSION_ID_SHIFT = 0;
+    static constexpr int CLONE_FIELD_LIST_REG_IDX = 1;
+    static constexpr uint64_t CLONE_FIELD_LIST_MASK = 0x00000000ffff0000;
+    static constexpr uint64_t CLONE_FIELD_LIST_SHIFT = 16;
+    static constexpr int LF_FIELD_LIST_REG_IDX = 1;
+    static constexpr uint64_t LF_FIELD_LIST_MASK = 0x0000ffff00000000;
+    static constexpr uint64_t LF_FIELD_LIST_SHIFT = 32;
+    static constexpr int RESUBMIT_FLAG_REG_IDX = 1;
+    static constexpr uint64_t RESUBMIT_FLAG_MASK = 0xffff000000000000;
+    static constexpr uint64_t RESUBMIT_FLAG_SHIFT = 48;
+
+    static constexpr int RECIRCULATE_FLAG_REG_IDX = 2;
+    static constexpr uint64_t RECIRCULATE_FLAG_MASK = 0x000000000000ffff;
+    static constexpr uint64_t RECIRCULATE_FLAG_SHIFT = 0;
+
+    static void clear_all(Packet *pkt) {
+        // except do not clear packet length
+        pkt->set_register(1, 0);
+        pkt->set_register(2, 0);
+    }
+    static uint16_t get_clone_mirror_session_id(Packet *pkt) {
+        uint64_t rv = pkt->get_register(CLONE_MIRROR_SESSION_ID_REG_IDX);
+        return static_cast<uint16_t>((rv & CLONE_MIRROR_SESSION_ID_MASK) >>
+                                     CLONE_MIRROR_SESSION_ID_SHIFT);
+    }
+    static void set_clone_mirror_session_id(Packet *pkt,
+                                            uint16_t mirror_session_id) {
+        uint64_t rv = pkt->get_register(CLONE_MIRROR_SESSION_ID_REG_IDX);
+        rv = ((rv & ~CLONE_MIRROR_SESSION_ID_MASK) |
+              (((uint64_t) mirror_session_id) <<
```

Comment:
There are still some C-style casts, I'm actually surprised the style checker is not complaining.

Reply:
Fixed the ones I found in this file.  Don't know anything about the style checker behavior.